app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -130,6 +149,14 @@ public void add(Message message) {
                 View messageView = LayoutInflater.from(parent.getContext()).inflate(R.layout.message_tile, parent, false);
                 MessageHolder messageHolder = new MessageHolder(messageView);
                 return messageHolder;
+            case VIEWTYPE_FOOTER:
+                footerView = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_loading, parent, false);
+                return new LoadingHolder(footerView);
+            case VIEWTYPE_HEADER:
+                headerView = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_loading, parent, false);
+                LoadingHolder headerLoadingHolder = new LoadingHolder(headerView);
+                isHeaderShowing(false);
+                return headerLoadingHolder;
```

Comment:
default case won't be needed because in getItemViewType(int position) function we are throwing an exception if the viewType is not known!
https://github.com/kunall17/zulip-android/blob/patch-9-recyclerview/app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java#L181-L183